lib/sles4sap/azure_cli.pm

```diff
@@ -3,40 +3,64 @@
 # Copyright SUSE LLC
 # SPDX-License-Identifier: FSFAP
 # Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Library wrapper around some az cli commands.
 
 package sles4sap::azure_cli;
 use strict;
-use warnings;
+use warnings FATAL => 'all';
 use testapi;
 use Carp qw(croak);
 use Exporter qw(import);
-use Regexp::Common qw(net);
+use Mojo::JSON qw(decode_json);
+
 
 =head1 SYNOPSIS
 
 Library to compose and run Azure cli commands
 =cut
 
 our @EXPORT = qw(
+  az_version
   az_group_create
+  az_group_name_get
   az_network_vnet_create
   az_network_nsg_create
   az_network_nsg_rule_create
   az_network_publicip_create
+  az_network_publicip_get
   az_network_lb_create
+  az_network_lb_probe_create
+  az_network_lb_rule_create
   az_vm_as_create
   az_vm_create
   az_vm_openport
+  az_vm_wait_cloudinit
   az_nic_id_get
   az_nic_name_get
   az_ipconfig_name_get
+  az_ipconfig_update
+  az_ipconfig_pool_add
 );
 
 
+=head2 az_version
+
+    az_version();
+
+Print the version of the az cli available on system
+=cut
+
+sub az_version {
+    assert_script_run('az --version');
```

Comment:
The intention is to print output in serial console only ?

Reply:
was to debug a problem with `az network public-ip create` http://openqaworker15.qa.suse.cz/tests/281895/logfile?filename=serial_terminal.txt

```
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.11/logging/__init__.py", line 1110, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "/root/.venv_az/lib64/python3.11/site-packages/knack/log.py", line 62, in format
    msg = logging.StreamHandler.format(self, record)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/logging/__init__.py", line 953, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/logging/__init__.py", line 687, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/logging/__init__.py", line 377, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
```

seems a problem related to the specific version of az included in the specific version of the az cli used in the pc_tools.qcow2 image I'm using (and in the latest qcow too). Issue is not present in old and in latest az cli version.

This API is just to have a way to see with version is in use. An different approach is to look at in this file  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/data/publiccloud/venv/az.txt and also to know which version of this file was used to generate the qcow2

Reply:
Ah ok I see. I was rather thinking about having output in result page vs. only in console. Not really an issue, only question.